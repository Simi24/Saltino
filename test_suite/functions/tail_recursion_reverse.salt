// Test di reverse con tail recursion (simulato con numeri)
def main() {
    // Simula reverse di [1,2,3,4,5] -> calcola 54321
    return reverse_number_tail(12345, 0)
}

def reverse_number_tail(num, acc) {
    if (num == 0) {
        return acc
    } else {
        return reverse_number_tail(num / 10, acc * 10 + num % 10)
    }
}
