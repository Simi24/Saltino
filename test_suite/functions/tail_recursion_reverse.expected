Program result: 54321
